docs/reproducibility.md

```diff
@@ -8,48 +8,152 @@ The assumption here is that the original build was done using a Docker.
 
 You need a couple pieces of information to begin with:
 
-- The ledger certificate. It might be distributed in a variety of ways by the ledger operator, please follow their guidance. Otherwise it is accessible at `https://<LEDGER-URL>/app/parameters`.
+- The ledger certificate. It might be distributed in a variety of ways by the ledger operator, please follow their guidance. Otherwise it is accessible at `https://<LEDGER-URL>/app/parameters`, e.g.:
 
-- The quote of a running application code, get it from `https://<LEDGER-URL>/node/quotes/self` (replace <LEDGER-URL> with the URL of your ledger), e.g.:
+    ```sh
+    $ curl -k "https://<LEDGER-URL>/app/parameters" | jq -r .serviceCertificate | base64 -d > cacert.der
+    $ openssl x509 -inform der -in cacert.der > cacert.pem
+    ```
+
+- The quote of a running application code, get it from `https://<LEDGER-URL>/node/quotes` which will contain the measurements of each node in the network. They will be the same almost all of the time except when upgrading to the new version e.g.:
 
     ```sh
-    $ curl -s --cacert cacert.pem https://<LEDGER-URL>/node/quotes/self > node-quote.json
+    $ curl -s --cacert cacert.pem https://<LEDGER-URL>/node/quotes > node-quotes.json
     ```
+
 - Source code version which was used to build the application, usually found in `https://<LEDGER-URL>/app/version`. If `app/version` is ambiguous then check the build logs of the Docker image to understand which commit produced the candidate value. e.g., `fb2c496416fbab20837fedda0ba6db58d819fa5f5c1b3916062eb2fb9d889966` was built from `fd77c0c69ee890bdc2fcf6ef0c9dddb7b211e164`.
 
     ```sh
     $ curl -s --cacert cacert.pem https://<LEDGER-URL>/app/version | jq ".version"
-    "1.2.3"
+    "0.12.3-0-gaaaaaaa"
     ```
 
-### Extract measurements
+- Security policy used to verify the container image ([ccf docs](https://microsoft.github.io/CCF/main/governance/gov_api_schemas/2024-07-01.html#get--gov-service-join-policy)), it will contain image layers, e.g.:
```

Comment:
```suggestion
- Join policy used to verify the container image ([ccf docs](https://microsoft.github.io/CCF/main/governance/gov_api_schemas/2024-07-01.html#get--gov-service-join-policy)), it will contain image layers, e.g.:
```

The join policy is what the primary applies to a joiner's attestation to decide if it can join or not. It does not directly contain container layer digests, rather it contains a set of digests of confidential container security policies that are acceptable (`snp.hostData`). Details can be found here: https://microsoft.github.io/CCF/main/governance/gov_api_schemas/2024-07-01.html#get--gov-service-join-policy

The confidential container security policies themselves do contain the digests of layers.

Reply:
The layers are in the policy, and this was exactly why I was interested in the policy. The fact that we have a join policy is secondary to the ability to get the policy to compare to the host_data measurement. If you are saying this is not guaranteed then there needs to be another API to get the security policy as we cannot regenerate it due to the complexity behind it.

Reply:
It is a good topic to discuss in a call, but let's release the changes to be able to have stable docker images and these steps that seem to work for now.